scripts/lib/api/generateApiComponents.test.ts

```diff
@@ -0,0 +1,224 @@
+// This code is a Qiskit project.
+//
+// (C) Copyright IBM 2024.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+import { expect, test } from "@jest/globals";
+import { CheerioAPI, Cheerio, load as cheerioLoad } from "cheerio";
+
+import {
+  ComponentProps,
+  prepareGitHubLink,
+  htmlSignatureToMd,
+  addExtraSignatures,
+  createOpeningTag,
+} from "./generateApiComponents";
+
+class Doc {
```

Comment:
Should probably deduplicate with `processHtml`. I think there can be issues with test files importing other test files. If so, you can define a new `lib/testUtils.ts`. Maybe call this type `CheerioDoc`.